File: aegis/src/main/java/com/gradientgeeks/aegis/sfe/dto/DeviceRegistrationResponse.java
package com.gradientgeeks.aegis.sfe.dto;

public class DeviceRegistrationResponse {
    
    private String deviceId;
    private String secretKey;
    private String status;
    private String message;
    
    public DeviceRegistrationResponse() {}
    
    public DeviceRegistrationResponse(String deviceId, String secretKey) {
        this.deviceId = deviceId;
        this.secretKey = secretKey;
        this.status = "success";
        this.message = "Device registered successfully";
    }
    
    public static DeviceRegistrationResponse error(String message) {
        DeviceRegistrationResponse response = new DeviceRegistrationResponse();
        response.status = "error";
        response.message = message;
        return response;
    }
    
    public static DeviceRegistrationResponse success(String deviceId, String secretKey) {
        return new DeviceRegistrationResponse(deviceId, secretKey);
    }
    
    public String getDeviceId() {
        return deviceId;
    }
    
    public void setDeviceId(String deviceId) {
        this.deviceId = deviceId;
    }
    
    public String getSecretKey() {
        return secretKey;
    }
    
    public void setSecretKey(String secretKey) {
        this.secretKey = secretKey;
    }
    
    public String getStatus() {
        return status;
    }
    
    public void setStatus(String status) {
        this.status = status;
    }
    
    public String getMessage() {
        return message;
    }
    
    public void setMessage(String message) {
        this.message = message;
    }
    
    @Override
    public String toString() {
        return "DeviceRegistrationResponse{" +
                "deviceId='" + deviceId + '\'' +
                ", secretKey='[REDACTED]'" +
                ", status='" + status + '\'' +
                ", message='" + message + '\'' +
                '}';
    }
}

File: sfe/sfe-client/src/main/java/com/gradientgeeks/aegis/sfe_client/model/DeviceRegistrationResponse.kt
package com.gradientgeeks.aegis.sfe_client.model

import com.google.gson.annotations.SerializedName

/**
 * Response model for device registration from the Aegis Security API.
 * 
 * Contains the device credentials that are securely provisioned during
 * the registration process.
 */
data class DeviceRegistrationResponse(
    
    /**
     * Unique device identifier assigned by the Aegis Security API.
     * This ID is used to identify the device in subsequent API calls.
     */
    @SerializedName("deviceId")
    val deviceId: String,
    
    /**
     * Cryptographically secure secret key for HMAC signing.
     * This key should be immediately stored in Android Keystore for security.
     */
    @SerializedName("secretKey")
    val secretKey: String,
    
    /**
     * Registration status indicator.
     * Typically "success" for successful registrations.
     */
    @SerializedName("status")
    val status: String,
    
    /**
     * Additional message from the server (optional).
     * May contain information about the registration process.
     */
    @SerializedName("message")
    val message: String? = null
)